### PR TITLE
Adding feature tests and steps for registration

### PR DESCRIPTION
**Pull request by danmalmx**
## PT Link

https://www.pivotaltracker.com/story/show/160721218

## Included in PR:
* Added feature test for registration
* Renamed feature testing files
* Reorganised feature steps

### CODE REVIEW

**Review by tochman — COMMENTED** [review]
Please revisit the naming of your files. We've been discussing this in several meetings and code reviews. Check with your team.

File: features/registration.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Feature: Registration
+    As a user
+    In order to be able to register for an account
+    I would like to be able to provide my details on the register page
+
+    Scenario: Successfull registration as a new user
+        Given That I am on the landing page
+        And I click on the "Login" link
```
Comment (magnus-thor) [inline features/registration.feature]: Any reason why you are clicking the login button first?

File: app/views/partials/_navbar.html.haml [hunk 2]
```diff
@@ -5,6 +5,6 @@
                 %span.font-semibold.text-2xl.tracking-tight FeedMe
                 .w-full.block.flex-grow.lg:flex.lg:items-center.lg:w-auto
                 .text-sm.lg:flex-grow
-                    %a.px-4.py.2.block.mt-8.lg:float-right.lg:mt-0.text-white.hover:text-grey{href: new_user_session_path}  Login
+                    %a.px-4.py.2.block.mt-8.lg:float-right.lg:mt-0.text-white.hover:text-grey{href: new_user_session_path}  Login/Register
```
Comment (tochman) [inline app/views/partials/_navbar.html.haml]: Please use `link_to` helper

File: app/views/partials/_navbar.html.haml [hunk 3]
```diff
@@ -5,6 +5,4 @@
                 %span.font-semibold.text-2xl.tracking-tight FeedMe
                 .w-full.block.flex-grow.lg:flex.lg:items-center.lg:w-auto
                 .text-sm.lg:flex-grow
-                    %a.px-4.py.2.block.mt-8.lg:float-right.lg:mt-0.text-white.hover:text-grey{href: new_user_session_path}  Login
-                    %a.px-4.py.2.block.mt-8.lg:float-left.lg:mt-0.text-white.hover:text-grey.px-4{href: "#"}  Food
-                    %a.px-4.py.2.block.mt-8.lg:float-left.lg:mt-0.text-white.hover:text-grey{href: "#"}  Drinks
\ No newline at end of file
+                %a.px-4.py.2.block.mt-8.lg.float-right.lg:mt-0.text-white.hover:text-grey= link_to 'Login/Register', new_user_session_path
```
Comment (tochman) [inline app/views/partials/_navbar.html.haml]: This is not the proper way to use the link_to helper. Please do some research by going over the official rails documentation.

**Review by tochman — COMMENTED** [review]
Close but not quite there yet.

File: features/products_listed_on_landing_page.feature [hunk 4]
```diff
@@ -17,5 +17,6 @@ Feature: List products
     Scenario: Product successfully shown on landing page
         Given That I am on the landing page
         Then I should see "Meatballs"
+        And then I wait 2 seconds
```
Comment (rupertlion) [inline features/products_listed_on_landing_page.feature]: Take  this step away

File: features/step_definitions/basic_steps.rb [hunk 5]
```diff
@@ -12,4 +12,8 @@
 
 Then("I fill in {string} with {string}") do |field, content|
     fill_in field, with: content    
+end
+
+Then("then I wait {int} seconds") do |seconds|
+    sleep seconds
```
Comment (rupertlion) [inline features/step_definitions/basic_steps.rb]: Take  this step away

**Pull request closed without merging**